shared/src/commonMain/kotlin/com/kcc/kmmhackathon/shared/network/MealApi.kt

```diff
@@ -62,6 +62,14 @@ class MealApi {
         }
     }
 
+    suspend fun reserveAMeal(id: String): Meal {
+        var meal = getMeal(id)
+        if (meal.quantity > 0) {
+            return patchMeal(id, meal.quantity - 1)
+        }
+        return meal
+    }
+
```

Comment:
If for example a user tries to reserve a meal that is available on their device but in actuality the quantity is at zero (because they didn't refresh the list), then we are just returning the meal back. 
So _I think_ then a toast would be displayed with a collection code?  Because it's technically success state.

We could make the return optional and return null if we don't patch and then catch that?
Or maybe this functionality would be better off being in the MealsSDK (and same for the function that sorts and filters all the meals too 🤔 ) and we could throw an error from there? 
We can't do the same as what I did on ios because when the meal is returned with 0 for quantity, we can't distinguish between it being zero because before it was 1, or it's zero because it was zero all along and we didn't make the PATCH.

Anyway I'm just rambling and putting my thoughts down